src/panels/KaitoModelsPanel.ts

```diff
@@ -0,0 +1,295 @@
+            resetStateRequest: false,
+            cancelRequest: false,
+        };
+    }
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            generateCRDRequest: (params) => {
+                this.handleGenerateCRDRequest(params.model);
+            },
+            deployKaitoRequest: (params) => {
+                this.handleDeployKaitoRequest(params.model, params.yaml, webview);
+            },
+            workspaceExistsRequest: (params) => {
+                this.handleWorkspaceExistsRequest(params.model);
+            },
+            updateStateRequest: (params) => {
+                this.handleUpdateStateRequest(params.model, webview);
+            },
+            resetStateRequest: () => {
+                this.handleResetStateRequest(webview);
+            },
+            cancelRequest: () => {
+                this.cancel();
+            },
+        };
+    }
+
+    private async handleGenerateCRDRequest(yaml: string) {
+        const doc = await vscode.workspace.openTextDocument({
+            content: yaml,
+            language: "yaml",
+        });
+        vscode.window.showTextDocument(doc);
+    }
+
+    nullIsFalse(value: boolean | null): boolean {
+        if (value === null) {
+            return false;
+        } else {
+            return value;
+        }
+    }
+
+    private async handleDeployKaitoRequest(model: string, yaml: string, webview: MessageSink<ToWebViewMsgDef>) {
+        this.cancelToken = false;
+        try {
+            const tempFilePath = join(tmpdir(), `kaito-deployment-${Date.now()}.yaml`);
+            writeFileSync(tempFilePath, yaml, "utf8");
+            const command = `apply -f ${tempFilePath}`;
+            const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+            if (failed(kubectlresult)) {
+                // Can add additional error handling here
+                vscode.window.showErrorMessage(
+                    `There was an error with deployment. Try deploying again. ${kubectlresult.error}`,
+                );
+                return;
+            }
+        } catch (error) {
+            vscode.window.showErrorMessage(`Error during deployment: ${error}`);
+        }
+        this.handleUpdateStateRequest(model, webview);
+        let progress = await this.getProgress(model);
+        // vscode.window.showInformationMessage(`initial workspace rdy: ${progress.workspaceReady}`); - debug 2
+        // vscode.window.showInformationMessage(
+        //     `PROGRESSVALUES1 - Resource Ready: ${progress.resourceReady}, Inference Ready: ${progress.inferenceReady}, Workspace Ready: ${progress.workspaceReady}, Age: ${progress.age}`,
+        // ); - debug 3
+        while (!this.nullIsFalse(progress.workspaceReady) && !this.cancelToken) {
+            // vscode.window.showInformationMessage(this.cancelToken.toString()); - debug 4
```

Comment:
💡 Observation: shall we remove these commented out code or is this specifically for any use latter?